fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -245,24 +245,45 @@ void applyMetadataMutations(SpanID const& spanContext,
 					}
 				}
 			} else if (m.param1.startsWith(tssMappingKeys.begin)) {
+				// Normally uses key backed map, so have to use same unpacking code here.
+				UID ssId = Codec<UID>::unpack(Tuple::unpack(m.param1.removePrefix(tssMappingKeys.begin)));
+				UID tssId = Codec<UID>::unpack(Tuple::unpack(m.param2));
 				if (!initialCommit) {
 					txnStateStore->set(KeyValueRef(m.param1, m.param2));
-					if (tssMapping) {
```

Comment:
The CommitProxy bug here was that the update to the in-memory TSS mapping was inside the if (!initialCommit) block, so it would only execute it for new transactions, not when recovering the transaction state store into a new proxy. This meant that every recovery or new commit proxy would lose the mapping.